pkg/generic/thrift/http.go

```diff
@@ -77,8 +108,52 @@ func (r *ReadHTTPResponse) SetBase64Binary(enable bool) {
 	r.base64Binary = enable
 }
 
+// SetEnableDynamicgo enable/disable dynamicgo decoding.
+func (r *ReadHTTPResponse) SetEnableDynamicgo(enable bool, opts conv.Options, msg remote.Message) {
+	r.enableDynamicgo = enable
+	r.opts = opts
+	r.msg = msg
+}
+
 // Read ...
 func (r *ReadHTTPResponse) Read(ctx context.Context, method string, in thrift.TProtocol) (interface{}, error) {
+	if r.enableDynamicgo {
+		tProt, ok := in.(*cthrift.BinaryProtocol)
+		if !ok {
+			return nil, perrors.NewProtocolErrorWithMsg("TProtocol should be BinaryProtocol")
+		}
+
+		if r.svc.DynamicgoDsc == nil {
+			return nil, perrors.NewProtocolErrorWithMsg("svcDsc.DynamicgoDsc is nil")
+		}
+		tyDsc := r.svc.DynamicgoDsc.Functions()[method].Response()
+
+		msgBeginLen := bthrift.Binary.MessageBeginLength(method, thrift.TMessageType(r.msg.MessageType()), r.msg.RPCInfo().Invocation().SeqID())
+		transBuf, err := tProt.ByteBuffer().ReadBinary(r.msg.PayloadLen() - msgBeginLen - bthrift.Binary.MessageEndLength())
+		if err != nil {
+			return nil, err
+		}
+
+		buf := make([]byte, 0, len(transBuf)*2)
+		resp := descriptor.NewHTTPResponse()
+		ctx = context.WithValue(ctx, conv.CtxKeyHTTPResponse, resp)
+		cv := t2j.NewBinaryConv(r.opts)
+		// decode with dynamicgo
+		// thrift []byte to json []byte
+		if err = cv.DoInto(ctx, tyDsc, transBuf, &buf); err != nil {
+			return nil, err
+		}
+
+		// unwrap one level of json string
+		if len(buf) > structWrapLen {
+			secondBracket := strings.Index(string(buf)[1:], "{") + 1
```

Comment:
What does this code do?

Reply:
@jayantxie In original read method, the resp body will be a map, which doesn't include the top level. For example, the map is not {"xxx" : {"aa": "xx", "bb": "yy"...}} but {"aa": "xx", "bb": "yy"...}. But the buf will be like {"xxx" : {"aa": "xx", "bb": "yy"...}} after doing dynamicgo conversion, so I want to unwrap the top level here.

Reply:
no need this code anymore